### PR TITLE
Added controls for the users to manually feed the ray tracing acceleration structure that should be used for a given camera (case 1370678).

### PR DESCRIPTION
**Pull request by anisunity**
https://fogbugz.unity3d.com/f/cases/1370678/
image

Testing status
I need to add a test that creates a ray tracing acceleration structure and feeds it manually to the camera.
Made sure that when the user didn't feed an acceleration structure the ray tracing effects were not executed.
Made sure that everything still worked as expected when in automatic mode.

### CODE REVIEW

**Review by remi-chapelain — CHANGES_REQUESTED** [review]
Fixed a few things:
- Changed the property name to Acceleration Structure Build Mode to avoid confusion with the act of building a player
- Put the new property in additional properties section
- Updated the tooltip and the doc with a little bit more information
- Tested with other RT effects
--- RTGI ✔️ 
--- RTAO ✔️ 
--- SubSurfaceScattering ✔️ 
--- Pathtracer - Some issues 
--- Recursive Rendering - Some issues
--- Ray Traced Shadows - Crash
- Added a new section in documentation with a snippet example on how to properly feed manually built RTAS to the camera
- Added @Vic-Cooper as reviewer for the new added section in documentation

image

Now about issues:
- Found a crash when using ray traced shadow in manual mode. ⚔️ 
- Also, layer mask (in RTR, RTGI, RTAO, PAthtracer) does not work anymore, I guess it makes sense but maybe it could be interesting to note it somewhere that it's intrinsically not compatible with lights layer since it's users who feeds the objects. 
- Also, with pathtracer and custom RTAS, you get unexpected results like losing secondary and more bounces for some reasons. 
- Objects in recursive rendering pass not passed in the RTAS are black, so there's no fallback to raster for recursive rendering.

**Comment by anisunity** [comment]
i'll be checking the crashes today

**Comment by anisunity** [comment]
The PR https://github.com/Unity-Technologies/Graphics/pull/5988 needs to be merged first before this PR gets merged.

**Comment by anisunity** [comment]
yes i started updating it

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs [hunk 1]
```diff
@@ -1541,6 +1541,7 @@ void PreprocessVisibleLights(CommandBuffer cmd, HDCamera hdCamera, in CullingRes
                 m_ProcessedLightsBuilder.Build(
                     hdCamera,
                     cullResults,
+                    GetRayTracingState(),
```
[the remark this answers is not in the document]
Comment (kecho) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs:1545]: LGTM

**Review by kecho — APPROVED** [review]

File: com.unity.render-pipelines.high-definition/Documentation~/Ray-Tracing-Settings.md [hunk 2]
```diff
@@ -7,6 +7,63 @@ In the High Definition Render Pipeline (HDRP), various ray-traced effects share
 1. In the Scene or Hierarchy view, select a GameObject that contains a Volume component to view it in the Inspector.
 2. In the Inspector, navigate to **Add Override > Ray Tracing** and click on **Ray Tracing Settings**.
 
+## Mannually building the Ray Tracing Acceleration Structure
+
+HDRP provides a utility function that adds objects to the ray tracing acceleration structure.
```
Comment (remi-chapelain) [inline com.unity.render-pipelines.high-definition/Documentation~/Ray-Tracing-Settings.md:12]: This was the previous version and needs to be updated

**Review by remi-chapelain — CHANGES_REQUESTED** [review]
Tested all ray traced effects independently and everything is working as it should, even though it's not what the user would necessarily expect when some object are not included in the RTAS. ✔️ 

To avoid this confusion, I would like the documentation to include the fact that if you are using a ray traced effect on an object (ray traced shadow, RTAO or RTR for example) and this object is not included in the RTAS, because of how the ray tracing pipeline is made, the effect won't fallback to its rasterization counterpart. 
Meaning that instead of raytraced shadow you'll have no shadows, instead of RTR, you won't have SSR, you will have no reflections.. etc. 
An image from the test could help illustrate this result !

**Comment by anisunity** [comment]
Fixed test, relaunching dxr tests and then merging

**Pull request merged**